SCAutolib/models/user.py

```diff
@@ -109,61 +173,149 @@ def cnf(self):
+        :type pin: str
         :param cnf: CNF file to be associated with the user
-        :type Path
+        :type cnf: Path
         :param key: Key to be associated with the user
-        :type Path
+        :type key: Path
         :param cert: Certificate to be associated with the user.
-        :type Path
+        :type cert: Path
         """
 
-        super().__init__(username, password, pin, card, cnf, key, cert)
+        super().__init__(*args, **kwargs)
         self._meta_client = ipa_server.meta_client
-
-        self._meta_client.login("admin", ipa_server._ipa_server_admin_passwd)
+        self._ipa_hostname = ipa_server.ipa_server_hostname
 
     def add_user(self):
-        r = self._meta_client.user_add(self.username, self.username,
-                                       self.username, self.username,
-                                       o_userpassword=self.password,
-                                       o_homedirectory=f"/home/{self.username}")
-        self.card_dir = f"/home/{self.username}"
-        logger.debug(r)
-        logger.info(f"User {self.username} is added to the IPA server")
+        """
+        Adds IPA user to IPA server.
+        """
+        try:
+            r = self._meta_client.user_add(self.username, self.username,
+                                           self.username, self.username,
+                                           o_userpassword=self.default_password)
+            logger.debug(r)
+
+            # To avoid forcing IPA server to change the password on first login
+            # we changing it through the client
+            client = python_freeipa.client.Client(self._ipa_hostname,
+                                                  verify_ssl=False)
+            client.change_password(self.username, self.password,
+                                   self.default_password)
+            logger.info(f"User {self.username} is added to the IPA server")
+        except python_freeipa.exceptions.DuplicateEntry:
+            msg = f"User {self.username} already exists on the " \
+                  f"IPA server. Username should be unique to avoid " \
+                  f"future problems with collisions"
+            logger.critical(msg)
+            raise SCAutolibException(msg)
 
     def delete_user(self):
-        r = self._meta_client.user_del(self.username)["result"]
-        logger.info(f"User {self.username} is removed from the IPA server")
-        run(['rm', '-r', '-f', f"/home/{self.username}"], check=True)
-        logger.info(f"User {self.username} directory is removed.")
-        logger.debug(r)
+        """
+        Deletes the user and user's card directory
+        """
+        try:
+            r = self._meta_client.user_del(self.username)["result"]
+            logger.info(f"User {self.username} is removed from the IPA server")
+            logger.debug(r)
+        except python_freeipa.exceptions.NotFound:
+            pass
+        if list(self.card_dir.iterdir()):
```

Comment:
Is this check needed? It basically boils down to "if `card_dir` is not empty, remove `card_dir` recursively", right?

So if it is empty, it remains there.

What about:
```
if self.card_dir.exists():
    rmtree(self.card_dir)
    ...
```

Reply:
makes sense